File: lib/models/song.dart
import 'package:hive/hive.dart';
import 'section.dart';

part 'song.g.dart';

@HiveType(typeId: 0)
class Song extends HiveObject {
  @HiveField(0)
  String id;

  @HiveField(1)
  String title;

  @HiveField(2)
  String? artist;

  @HiveField(3)
  String filePath;

  @HiveField(4)
  String? youtubeUrl;

  @HiveField(5)
  List<Section> sections;

  @HiveField(6)
  String notes;

  @HiveField(7)
  int durationMs;

  @HiveField(8)
  DateTime addedAt;

  Song({
    required this.id,
    required this.title,
    this.artist,
    required this.filePath,
    this.youtubeUrl,
    List<Section>? sections,
    this.notes = '',
    this.durationMs = 0,
    DateTime? addedAt,
  })  : sections = sections ?? [],
        addedAt = addedAt ?? DateTime.now();
}

File: transcreve-py/models/song.py
from dataclasses import dataclass, field
from datetime import datetime
import uuid

from .section import Section


@dataclass
class Song:
    title: str
    file_path: str
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    artist: str | None = None
    youtube_url: str | None = None
    sections: list[Section] = field(default_factory=list)
    notes: str = ""
    duration_ms: int = 0
    added_at: datetime = field(default_factory=datetime.now)
